src/main/scala/com/payalabs/scalajs/react/bridge/ReactBridgeComponent.scala

```diff
@@ -123,3 +125,11 @@ object ReactBridgeComponent {
     case value => value.asInstanceOf[js.Any]
   }
 }
+
+/**
+ *  extends this, to convert scala classes to js literals
+ */
+trait ReactBridgeObject extends Any {
+  def toJS : js.Dynamic
```

Comment:
I think returning `js.Any` may be sufficient here. Then your code will be simpler:
```
trait ReactBridgeObject {
  def toJS: js.Any
}

class NameType (val name :String) extends AnyVal with ReactBridgeObject {
  override def toJS  = name
}
```